messaging/src/main/java/org/axonframework/configuration/Component.java

```diff
@@ -132,12 +145,24 @@ public boolean matches(@Nonnull Identifier<?> other) {
          * otherwise.
          */
         public boolean matchesType(@Nonnull Identifier<?> other) {
-            return type.isAssignableFrom(other.type());
+            return type.getTypeAsClass().isAssignableFrom(other.type().getTypeAsClass());
+        }
+
+        /**
+         * Returns the raw type of the {@link #type()}.
+         * <p>
+         * This means any generics present on the {@code type} are lost. When those are needed, be sure to use
+         * {@link #type()} instead.
+         *
+         * @return The raw type of the {@link #type()}.
+         */
+        public Class<C> rawType() {
```

Comment:
I think `rawType` is the wrong name here. `typeAsClass()` would be better.
But if you think about it, the `Type` is the raw type.

Reply:
You're right here. I'll rename to `typeAsClass`